packages/valory/skills/decision_maker_abci/behaviours/reedem.py

```diff
@@ -448,11 +448,11 @@ def _check_already_redeemed(self) -> WaitableConditionType:
             if not result:
                 n_retries += 1
                 keep_fraction = (1 - self.params.reduce_factor) ** n_retries
-                batch_size = int(self.params.event_filtering_batch_size * keep_fraction)
-                msg = (
+                batch_size = int(batch_size * keep_fraction)
```

Comment:
Previously this worked because we used the non-incremented stored value. To keep this, you need to adjust keep_fraction to be constant above.